kubernetes-deployment/maintainlib/common.py

```diff
@@ -79,12 +79,12 @@ def write_generated_file(generated_file, file_path):
 
 def sftp_paramiko(src, dst, filename, host_config):
 
-    hostip = host_config['hostip']
-    username = host_config['username']
-    password = host_config['password']
+    hostip = str(host_config['hostip'])
+    username = str(host_config['username'])
+    password = str(host_config['password'])
     port = 22
     if 'sshport' in host_config:
-        port = host_config['sshport']
+        port = int(host_config['sshport'])
```

Comment:
>port [](start = 8, length = 4)

please check whether port value is valid.

Reply:
Another feature, do it in the feature in our plan.

Reply:
By the way, I will remove most function in bootstrap. And instead, I will call them from maintainlib.common.

Reply:
so you prefer to do ut after the code refactoring? 

---
In reply to: [170410130](https://github.com/Microsoft/pai/pull/330#discussion_r170410130) [](ancestors = 170410130)

Reply:
Yes